src/components/SpectralProfiler/SpectralProfilerComponent.tsx

```diff
@@ -152,21 +155,59 @@ export class SpectralProfilerComponent extends React.Component<WidgetProps> {
+        if (spectralType === SpectralType.CHANNEL) {
+            // If value it channel, ie. Int, return integer value.
+            return toFixed(data);
+        } else {
+            // Determine proper rounding and trim trailing zeros from return value.
+            return toFormattedNotation(data, diff);
+        }
+    };
+
+    private genCursoInfoString = (data: Point2D[], cursorXValue: number, cursorXUnit: string, label: string, start: number, end: number): string => {
+        const frame = this.widgetStore.effectiveFrame;
+
+        let diffLeft: number = undefined;
+        let secondaryXUnit = "";
+        let channelString = "";
+        let cursorInfoString: string = "";
+
         const nearest = binarySearchByX(data, cursorXValue);
+
         if (nearest?.point && nearest?.index >= 0 && nearest?.index < data?.length) {
-            let floatXStr = "";
-            const diffLeft = nearest.index - 1 >= 0 ? Math.abs(nearest.point.x - data[nearest.index - 1].x) : 0;
-            if (diffLeft > 0 && diffLeft < 1e-6) {
-                floatXStr = formattedNotation(nearest.point.x);
-            } else if (diffLeft >= 1e-6 && diffLeft < 1e-3) {
-                floatXStr = toFixed(nearest.point.x, 6);
+            let primaryXStr: string = "";
+
+            // We calculate the difference between neighboring values to get and estimate of the precision.
+            diffLeft = nearest.index - 1 >= 0 ? Math.abs(data[nearest.index].x - data[nearest.index - 1].x) : Math.abs(data[nearest.index + 1].x - data[nearest.index].x);
+
+            // Use precision to determine the proper rounding and zero suppression for displayed value. Data and secondary
+            // are handled idfferently because they have different structures.
+            primaryXStr = this.precisionFormatting(nearest, data[nearest.index].x, diffLeft, frame.spectralType);
+
+            if (this.widgetStore.secondaryAxisCursorInfoVisible) {
+                let secondary = this.widgetStore.effectiveFrame.channelSecondaryValues.slice(start, end);
+
+                // Use precision to determine the proper rounding and zero suppression for displayed value. Data and secondary
+                // are handled idfferently because they have different structures.
+                diffLeft = nearest.index - 1 >= 0 ? Math.abs(data[nearest.index].x - data[nearest.index - 1].x) : Math.abs(data[nearest.index + 1].x - data[nearest.index].x);
+
+                // Use precision to determine the proper rounding and zero suppression for displayed value.
+                const secondaryXStr = this.precisionFormatting(nearest, secondary[nearest.index], diffLeft, frame.spectralTypeSecondary);
+
+                if (frame.spectralTypeSecondary !== SpectralType.CHANNEL) secondaryXUnit = frame.spectralUnitSecondary;
+                else channelString = "Channel ";
+
+                const xLabel =
```

Comment:
Maybe we can apply `let xLabel = primary string` before if block and then `xLabel += secondary string` in the if block to avoid duplicate code.